File: _history/20260206/test/client_code/tests/client/echo.js
/*
client/echo.js
*/

export default async () => {
  await (async () => {
    const echo = await use("@@/echo/");
    const result = await echo({ foo: 42 }, 1, 2, 3);
    console.log("result:", result);
  })();

  await (async () => {
    const echo = await use("@@/echo/", { response: false });
    const result = await echo({ foo: 42 }, 1, 2, 3);
    console.log("result:", result);
  })();

  await (async () => {
    const ping = await use("@@/echo:ping");
    const result = await ping();
    console.log("result:", result);
  })();

};

File: _scratch/echo.py
import json
from pathlib import Path
from anvil.server import (
    call,
    connect,
   
)

UTF_8 = "utf-8"


from anvil.server import call, connect
connect(
        (json.loads((Path.cwd() / "secrets.json").read_text(encoding=UTF_8)))[
            "development"
        ]["client"]
    )


response = call('main', 'echo', request=dict(args=[42]), submission=0)

print('response:', response)

result = response['result']


result = json.dumps(result)

print('result:', result)